Source/Scene/ShadowMapShader.js

```diff
@@ -98,7 +194,20 @@ define([
         var bias = isPointLight ? shadowMap._pointBias : (isTerrain ? shadowMap._terrainBias : shadowMap._primitiveBias);
         var exponentialShadows = shadowMap._exponentialShadows;
 
-        fs = ShaderSource.replaceMain(fs, 'czm_shadow_main');
+        var defines = fs.defines.slice(0);
+        var sources = fs.sources.slice(0);
+
+        if (isTerrain) {
+            if (!terrainHasPositionVarying(defines)) {
+                defines.push('ENABLE_DAYNIGHT_SHADING');
+            }
+            hasNormalVarying = hasNormalVarying && terrainHasNormalVarying(defines);
+        }
```

Comment:
The receive-shadows fragment shader doesn't need to push any extra defines, since the `v_positionEC` and `v_normalEC` are freely available. Although it does make me wonder if we should always enable day-night shading when shadows are enabled to handle the lack of shadows during the night.